contribs/github-bot/comment.go

```diff
@@ -0,0 +1,276 @@
+	issue, ok := actionCtx.Event["issue"].(map[string]any)
+	if !ok {
+		return fmt.Errorf("unable to get issue on issue comment event")
+	}
+
+	num, ok := issue["number"].(float64)
+	if !ok || num <= 0 {
+		return fmt.Errorf("unable to get issue number on issue comment event")
+	}
+
+	// Check if change is only a checkbox being checked or unckecked
+	checkboxes := regexp.MustCompile(`(?m:^- \[[ x]\])`)
+	if checkboxes.ReplaceAllString(current, "") != checkboxes.ReplaceAllString(previous, "") {
+		// If not, restore previous comment body
+		if !gh.DryRun {
+			gh.SetBotComment(previous, int(num))
+		}
+		return fmt.Errorf("bot comment edited outside of checkboxes")
+	}
+
+	// Check if actor / comment editor has permission to modify changed boxes
+	currentChecks := getCommentManualChecks(current)
+	previousChecks := getCommentManualChecks(previous)
+	edited := ""
+	for key := range currentChecks {
+		if currentChecks[key][0] != previousChecks[key][0] {
+			// Get teams allowed to edit this box from config
+			var teams []string
+			found := false
+			_, manualRules := config(gh)
+
+			for _, manualRule := range manualRules {
+				if manualRule.Description == key {
+					found = true
+					teams = manualRule.Teams
+				}
+			}
+
+			// If rule were not found, return to reprocess the bot comment entirely
+			// (maybe bot config was updated since last run?)
+			if !found {
+				gh.Logger.Debugf("Updated rule not found in config: %s", key)
+				return nil
+			}
+
+			// If teams specified in rule, check if actor is a member of one of them
+			if len(teams) > 0 {
+				if gh.IsUserInTeams(actionCtx.Actor, teams) {
+					if !gh.DryRun {
+						gh.SetBotComment(previous, int(num))
+					}
+					return fmt.Errorf("checkbox edited by a user not allowed to")
+				}
+			}
+
+			// If box was checked
+			reg := regexp.MustCompile(fmt.Sprintf(`(?m:^- \[%s\] %s.*$)`, currentChecks[key][0], key))
+			if strings.TrimSpace(currentChecks[key][0]) == "x" {
+				replacement := fmt.Sprintf("- [%s] %s (checked by @%s)", currentChecks[key][0], key, actionCtx.Actor)
+				edited = reg.ReplaceAllString(current, replacement)
+			} else {
+				replacement := fmt.Sprintf("- [%s] %s", currentChecks[key][0], key)
+				edited = reg.ReplaceAllString(current, replacement)
+			}
+		}
+	}
+
+	// Update comment with username
+	if edited != "" && !gh.DryRun {
+		gh.SetBotComment(edited, int(num))
+		gh.Logger.Debugf("Comment manual checks updated successfully")
+	}
+
+	return nil
+}
+
+func updateComment(gh *client.GitHub, pr *github.PullRequest, content CommentContent) {
+	// Custom function to string markdown links
+	funcMap := template.FuncMap{
+		"stripLinks": func(input string) string {
+			reg := regexp.MustCompile(`\[(.*)\]\(.*\)`)
+			return reg.ReplaceAllString(input, "$1")
+		},
+	}
+
+	// Generate bot comment using template file
+	const tmplFile = "comment.tmpl"
+	tmpl, err := template.New(tmplFile).Funcs(funcMap).ParseFiles(tmplFile)
+	if err != nil {
+		panic(err)
```

Comment:
This can only happen if someone modified this code, removed the template file or something similar (so not because of any interaction with user or external event during the runtime), this seems to be a valid reason to panic for me (like when we use `MustCompile` for a regex.

Reply:
Modified to return an error up to the main 👍